### ISSUE TITLE
radarr custom formats alignment

### ISSUE THREAD

**Issue by Fuochi** [golift/starr#84]
@davidnewhall I was looking into radarr customformats methods. It would be great to align them with indexer/downloadclient/notification. However, I haven't made a PR for that, because there are few elements that I'm not sure how will impact your existing usage. Here few examples of changes that I'm not sure about:

- I don't understand the scope of this:
  https://github.com/golift/starr/blob/7260aa06b298e860a9182cab2e691377908f640a/radarr/customformat.go#L70-L72

- This I think can be omitted if we add the `omitempty` flag to `ID`, it shouldn't be a big deal:
    https://github.com/golift/starr/blob/7260aa06b298e860a9182cab2e691377908f640a/radarr/customformat.go#L74

- I would like to remove the `cfID` input from here, and always use `format.ID`, is there any case in which the 2 would differ from each other?
    https://github.com/golift/starr/blob/7260aa06b298e860a9182cab2e691377908f640a/radarr/customformat.go#L90-L98

- Also, to align the behaviour we should use 2 structs `CustomFormatInput` and `CustomFormatOutput`, and leverage the structs `starr.FieldInput` and `starr.FieldInput`. However, this would mean a lot of breaking changes for this endpoint.

In short, If you are already using this in some places, I think it would be better if you update this to be the least disruptive as possible. If you aren't using this much, I can make a PR and we can work on that.

**Comment by davidnewhall** [comment]
> I don't understand the scope of this:

looks like me being clever. doesn't need to stay.

> This I think can be omitted if we add the omitempty flag to ID, it shouldn't be a big deal:

Yeah, I think removing that line of code is fine. Although it has nothing to do with omitempty.

> Also, to align the behaviour we should use 2 structs CustomFormatInput and CustomFormatOutput,

I'm not aware of what's different between input and output on these?  I suspect it's the `fields`.

So long as the resulting json structures remain the same, Notifiarr should be fine. I can deal with code changes pretty easy. I've already had to do quite a few for the things we've done in this library.

**Comment by davidnewhall** [comment]
You can also jump on Discord if you want to chat about things. :)

**Comment by Fuochi** [comment]
> I'm not aware of what's different between input and output on these? I suspect it's the `fields`.

Correct! Also `implementationName` and `infoLink` can be treated as output only, in the same way we did for the other endpoints.
If you want, I can make a PR, and we can start looking at that.

Also, if we want to lower possible impacts, we can start working on sonarr `customFormat` instead, which is used by v4 version only. Then, we can validate the change there, and apply the same logic to radarr later.

What do you think?

**Comment by davidnewhall** [comment]
Notifiarr uses custom formats in radarr and sonarr v4. I'd appreciate if you do one at a time, and let me adapt my code over there. I'm all for making this library better, and happy to have your help with it. Thanks!

**Comment by davidnewhall** [comment]
Radarr changes seem OK in Notifiarr.

**Comment by davidnewhall** [comment]
Heads up I changed the default branch to `main` and replaced testing in Travis-CI with testing in GitHub Actions.
